app/lib/anonymizer.rb

```diff
@@ -0,0 +1,118 @@
+class Anonymizer
+  def self.anonymize_all_data!
+    ActiveRecord::Base.connection.tables.each do |table_name|
+      anonymize_table!(table_name)
+    end
+  end
+
+  def self.anonymize_user_data!
+    anonymize_table!(User)
+    anonymize_table!(Receipt)
+    anonymize_table!(Rdv)
+  end
+
+  def self.anonymize_table!(table_name)
+    new(table_name).anonymize_table!
+  end
+
+  def self.anonymize_record!(record)
+    new(record.class.table_name).anonymize_record!(record)
+  end
+
+  def initialize(table_name)
+    @table_name = table_name
+  end
+
+  def anonymize_record!(record)
+    record.class.where(id: record.id).update_all(anonymized_attributes) # rubocop:disable Rails/SkipsModelValidations
+    record.reload
+  end
+
+  def anonymize_table!
+    if Rails.env.production? && !ENV["ETL"] == "true"
+      raise "L'anonymisation en masse est désactivée en production pour éviter les catastrophes"
+    end
+
+    if @table_name.in?(AnonymizerRules::TRUNCATED_TABLES)
+      db_connection.execute("TRUNCATE #{ActiveRecord::Base.sanitize_sql(@table_name)}")
+      return
+    end
+
+    if unidentified_column_names.present?
+      raise "Les règles d'anonymisation pour les colonnes #{unidentified_column_names.join(' ')} de la table #{@table_name} n'ont pas été définies"
```

Comment:
J'aime bien utilise `inspect` dans ce genre de cas, ça affiche des valeurs telles qu'on est habituées à les voir.

```suggestion
      raise "Les règles d'anonymisation pour les colonnes #{unidentified_column_names.inspect} de la table #{@table_name.inspect} n'ont pas été définies"
```

> RuntimeError: Les règles d'anonymisation pour les colonnes ["first_name", "last_name"] de la table "users" n'ont pas été définies

Reply:
oui, mais ce `join(' ')` est trèèèèès pratique pour copier coller les messages d'erreur dans le fichier de règles et constituer ce fichier en tdd